packages/bpk-component-nudger/src/BpkConfigurableNudger.js

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable react/forbid-prop-types */
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow strict */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import BpkButton from 'bpk-component-button';
+import { withButtonAlignment } from 'bpk-component-icon';
+import MinusIcon from 'bpk-component-icon/sm/minus';
+import PlusIcon from 'bpk-component-icon/sm/plus';
+import { cssModules } from 'bpk-react-utils';
+
+import {
+  type CommonProps,
+  COMMON_PROP_TYPES,
+  COMMON_DEFAULT_PROPS,
+} from './common-types';
+import STYLES from './BpkNudger.scss';
+
+const getClassName = cssModules(STYLES);
+
+const AlignedMinusIcon = withButtonAlignment(MinusIcon);
+const AlignedPlusIcon = withButtonAlignment(PlusIcon);
+
+type Props<T> = {
+  ...$Exact<CommonProps<T>>,
+  inputClassName: ?string,
+  formatValue: T => string,
+  incrementValue: T => mixed,
```

Comment:
Why are we using `mixed` and not `T` for the return value? `mixed` is intended for use when the type is not known, but we infact know that the type here should be `T`